<?xml version="1.0" encoding="UTF-8"?>
<testng-results skipped="1" failed="1" total="2" passed="0">
  <reporter-output>
  </reporter-output>
  <suite name="Suite" duration-ms="22450" started-at="2024-12-15T17:54:38Z" finished-at="2024-12-15T17:55:01Z">
    <groups>
    </groups>
    <test name="SignUp and login" duration-ms="22450" started-at="2024-12-15T17:54:38Z" finished-at="2024-12-15T17:55:01Z">
      <class name="test.TC004_MenuValidation">
        <test-method status="PASS" signature="testDetails()[pri:0, instance:test.TC004_MenuValidation@69d9c55]" name="testDetails" is-config="true" duration-ms="1" started-at="2024-12-15T23:25:01Z" finished-at="2024-12-15T23:25:01Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testDetails -->
        <test-method status="SKIP" signature="menuValidation()[pri:0, instance:test.TC004_MenuValidation@69d9c55]" name="menuValidation" duration-ms="0" started-at="2024-12-15T23:25:01Z" data-provider="readData" finished-at="2024-12-15T23:25:01Z">
          <exception class="java.lang.RuntimeException">
            <message>
              <![CDATA[java.lang.NullPointerException]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.RuntimeException: java.lang.NullPointerException
	at org.testng.internal.MethodInvocationHelper.invokeDataProvider(MethodInvocationHelper.java:163)
	at org.testng.internal.Parameters.handleParameters(Parameters.java:430)
	at org.testng.internal.Invoker.handleParameters(Invoker.java:1276)
	at org.testng.internal.Invoker.createParameters(Invoker.java:991)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1081)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at org.testng.TestRunner.privateRun(TestRunner.java:782)
	at org.testng.TestRunner.run(TestRunner.java:632)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:366)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:361)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:319)
	at org.testng.SuiteRunner.run(SuiteRunner.java:268)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1244)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1169)
	at org.testng.TestNG.run(TestNG.java:1064)
	at org.testng.remote.RemoteTestNG.run(RemoteTestNG.java:113)
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:206)
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:177)
Caused by: java.lang.NullPointerException
	at org.apache.poi.xssf.usermodel.XSSFWorkbook.getSheet(XSSFWorkbook.java:1217)
	at utils.UtilityClass.readExcel(UtilityClass.java:82)
	at base.ProjectSpecifiedMethod.readData(ProjectSpecifiedMethod.java:53)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:86)
	at org.testng.internal.MethodInvocationHelper.invokeDataProvider(MethodInvocationHelper.java:137)
	... 20 more
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.RuntimeException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- menuValidation -->
      </class> <!-- test.TC004_MenuValidation -->
      <class name="test.TC001_SignIn">
        <test-method status="PASS" signature="reportInitialization()[pri:0, instance:test.TC001_SignIn@7f560810]" name="reportInitialization" is-config="true" duration-ms="314" started-at="2024-12-15T23:24:38Z" finished-at="2024-12-15T23:24:38Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- reportInitialization -->
        <test-method status="PASS" signature="setup()[pri:0, instance:test.TC001_SignIn@7f560810]" name="setup" is-config="true" duration-ms="1" started-at="2024-12-15T23:24:38Z" finished-at="2024-12-15T23:24:38Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setup -->
        <test-method status="PASS" signature="setup()[pri:0, instance:test.TC004_MenuValidation@69d9c55]" name="setup" is-config="true" duration-ms="0" started-at="2024-12-15T23:24:38Z" finished-at="2024-12-15T23:24:38Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setup -->
        <test-method status="PASS" signature="testDetails()[pri:0, instance:test.TC001_SignIn@7f560810]" name="testDetails" is-config="true" duration-ms="65" started-at="2024-12-15T23:24:38Z" finished-at="2024-12-15T23:24:38Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- testDetails -->
        <test-method status="PASS" signature="launchAndLoadURL(java.lang.String, java.lang.String)[pri:0, instance:test.TC001_SignIn@7f560810]" name="launchAndLoadURL" is-config="true" duration-ms="9170" started-at="2024-12-15T23:24:39Z" finished-at="2024-12-15T23:24:48Z">
          <params>
            <param index="0">
              <value>
                <![CDATA[chrome]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[https://www.bestbuy.com/?intl=nosplash]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- launchAndLoadURL -->
        <test-method status="FAIL" signature="signUpTest(java.lang.String, java.lang.String)[pri:0, instance:test.TC001_SignIn@7f560810]" name="signUpTest" duration-ms="10849" started-at="2024-12-15T23:24:48Z" data-provider="readData" finished-at="2024-12-15T23:24:59Z">
          <params>
            <param index="0">
              <value>
                <![CDATA[kssangeetha158@gmail.com]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[Sriramsangi@2224]]>
              </value>
            </param>
          </params>
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[expected [Sign In - Sign In Options - Best Buy] but found [Best Buy | Official Online Store | Shop Now & Save]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: expected [Sign In - Sign In Options - Best Buy] but found [Best Buy | Official Online Store | Shop Now & Save]
	at org.testng.Assert.fail(Assert.java:94)
	at org.testng.Assert.failNotEquals(Assert.java:513)
	at org.testng.Assert.assertEqualsImpl(Assert.java:135)
	at org.testng.Assert.assertEquals(Assert.java:116)
	at org.testng.Assert.assertEquals(Assert.java:190)
	at org.testng.Assert.assertEquals(Assert.java:200)
	at page.HomePage.validate_login(HomePage.java:49)
	at test.TC001_SignIn.signUpTest(TC001_SignIn.java:34)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:86)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:822)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1130)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at org.testng.TestRunner.privateRun(TestRunner.java:782)
	at org.testng.TestRunner.run(TestRunner.java:632)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:366)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:361)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:319)
	at org.testng.SuiteRunner.run(SuiteRunner.java:268)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1244)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1169)
	at org.testng.TestNG.run(TestNG.java:1064)
	at org.testng.remote.RemoteTestNG.run(RemoteTestNG.java:113)
	at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:206)
	at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:177)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- signUpTest -->
        <test-method status="PASS" signature="closebrowser()[pri:0, instance:test.TC001_SignIn@7f560810]" name="closebrowser" is-config="true" duration-ms="1058" started-at="2024-12-15T23:25:00Z" finished-at="2024-12-15T23:25:01Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- closebrowser -->
        <test-method status="PASS" signature="Close()[pri:0, instance:test.TC001_SignIn@7f560810]" name="Close" is-config="true" duration-ms="555" started-at="2024-12-15T23:25:01Z" finished-at="2024-12-15T23:25:01Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- Close -->
      </class> <!-- test.TC001_SignIn -->
    </test> <!-- SignUp and login -->
  </suite> <!-- Suite -->
</testng-results>
